internal/blocksync/pool.go

```diff
@@ -130,23 +130,22 @@ func (*BlockPool) OnStop() {}
 
 // spawns requesters as needed
 func (pool *BlockPool) makeRequestersRoutine(ctx context.Context) {
-	for {
-		if !pool.IsRunning() {
-			break
-		}
+	timer := time.NewTimer(0)
+	<-timer.C
+	defer timer.Stop()
 
+	for pool.IsRunning() {
 		_, numPending, lenRequesters := pool.GetStatus()
 		switch {
-		case numPending >= maxPendingRequests:
-			// sleep for a bit.
-			time.Sleep(requestIntervalMS * time.Millisecond)
-			// check for timed out peers
-			pool.removeTimedoutPeers()
-		case lenRequesters >= maxTotalRequesters:
-			// sleep for a bit.
-			time.Sleep(requestIntervalMS * time.Millisecond)
-			// check for timed out peers
-			pool.removeTimedoutPeers()
+		case numPending >= maxPendingRequests || lenRequesters >= maxTotalRequesters:
```

Comment:
I'd use an if statement with a continue rather than a switch statement with one case and a default.

Reply:
It's still unclear to me why there are so many `switch` statements throughout the codebase like this, where they could've been `if` statements. At first I was put off by the `switch` statements, but now they're starting to seem more readable. I imagine there are potentially some subtle performance differences between `switch` and `if` statements too, but they're probably negligible in this context.

Also some discussion on SO about this: https://stackoverflow.com/questions/97987/advantage-of-switch-over-if-else-statement

Reply:
I wouldn't worry so much about the performance case and I'd be surprised if they don't get compiled to the same thing, I think it comes down to readability, and therefore: 
- things that would be if/else (on one value) tend to make more sense as switch statements. "else" tends to be seen as a bit extra.
- there is a case where "contains" type operations are more clear as switch statements, and so they crop up in those cases. 
- otherwise, if statements are universally preferred.

Reply:
These seem like largely aesthetic and subjective considerations to me. I've changed it into an `if` statement, but personally I found the `switch` form more readable.